Enterprise-CMCS/managed-care-review
# MCR-4024: Fix cypress tests for linked rates

## Summary
[MCR-4024](https://jiraent.cms.gov/browse/MCR-4024)

- Added a new cypress test `'can unlock and resubmit a linked rate and change history updates'` in `unlockResubmit.spec`
- Updated `createAndSubmitContractWithRates` to use the new Contract and Rate API.
   - The function did not work using the HPP API to create rates, not sure how `rateReview.spec` even passed in other branches. 
   - Instead of trying to fix it with the old API, I updated it to use the new rates API.
   - I'm also returning `Contract` instead of `HealthPlanPackage`. 
      - This is because rate names differ between them and instead of updating HPP to match rate name formats today, I just switched to using `Contract` instead. 
      - This made `rateReview.spec` fail because it was looking for HPP rate names, but the rates were made using new Rate API with the new name format.
- Updated `rateReview.spec` to use `Contract` instead of HPP and update the tests now that rate names are less unique using the new rate name format.
- Fixed `fillOutLinkedRate` to look for the first rate option.
- Fixed `minnesotaStatePrograms` to use booleans for `isRateProgram` instead of string
   - Maybe this was causing `createAndSubmitContractWithRates` to fail.

#### Related issues

#### Screenshots

#### Test cases covered



## QA guidance

If it passes CI it should be good, but you could pull down and run both tests in local.
